[elasticsearchexporter] exporter panics when encoding non-string value scope attributes

### Component(s)

exporter/elasticsearch

### What happened?

## Description

The Elasticsearch exporter assumes all scope attribute values are strings, and will panic if they're not.

## Steps to Reproduce

Run the collector with the attached config. Yes, use a logs pipeline with the hostmetrics receiver -- originally an accident on my part, but it triggers the bug.

## Expected Result

The exporter should encode scope attribute values with the original type rather than coercing to a string.

## Actual Result

The exporter panics.

### Collector version

v0.119.0

### Environment information

## Environment

N/A

### OpenTelemetry Collector configuration

```yaml
receivers:                                                               
  hostmetrics:                                                           
    scrapers:                                                            
      cpu: {}                                                            
                                                                         
exporters:                                                               
  elasticsearch:                                                         
    endpoint: <snip>
    api_key: <snip>
                                                                         
service:                                                                 
  pipelines:                                                             
    logs:                                                             
      receivers: [hostmetrics]                                           
      processors: []                                                     
      exporters: [elasticsearch]
```

### Log output

```shell
panic: interface conversion: interface {} is bool, not string                                                                                                                                               
                                                                                                                                                                                                            
goroutine 1 [running]:                                                                                                                                                                                      
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.scopeToAttributes({0xc00075bab0?, 0xc001b03a4c?})                                                                  
        /home/andrew/projects/opentelemetry-collector-contrib/exporter/elasticsearchexporter/model.go:337 +0x27d                                                                                            
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.(*encodeModel).encodeLogDefaultMode(0xc001b02690, {0xc000dd6f78?, 0xc001b03a4c?}, {0xc000d91300?, 0xc001b03a4c?}, {0xc00075bab0?, 0xc001b03a4c?}, {{0x10a0d4fe,
 0x14}, {0x0, ...}, ...})                                                                                                                                                                                   
        /home/andrew/projects/opentelemetry-collector-contrib/exporter/elasticsearchexporter/model.go:136 +0x8e5           
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.(*encodeModel).encodeLog(0xc001b02690, {0xc000dd6f78?, 0xc001b03a4c?}, {0x0, 0x0}, {0xc000d91300?, 0xc001b03a4c?}, {0xc00075bab0?, 0xc001b03a4c?}, {0x0, ...}, 
...)                                                                                                                          
        /home/andrew/projects/opentelemetry-collector-contrib/exporter/elasticsearchexporter/model.go:113 +0x166           
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.(*elasticsearchExporter).pushLogRecord(0xc000bc8540, {0x120d3368, 0xc00075bb20}, {0xc000dd6f78?, 0xc001b03a4c?}, {0x0, 0x0}, {0xc000d91300?, 0xc001b03a4c?}, {0
xc00075bab0, ...}, ...)                                                                                                                 
        /home/andrew/projects/opentelemetry-collector-contrib/exporter/elasticsearchexporter/exporter.go:187 +0x54b          
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.(*elasticsearchExporter).pushLogsData(0xc000bc8540, {0x120d3368, 0xc00075bb20}, {0xc0011a3f08?, 0xc001b03a4c?})
        /home/andrew/projects/opentelemetry-collector-contrib/exporter/elasticsearchexporter/exporter.go:138 +0xb37                    
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsRequest).Export(0xc000d81d60, {0x120d3368, 0xc00075bb20})        
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.119.0/exporterhelper/logs.go:64 +0x83            
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*TimeoutSender).Send(0xc000f645a0, {0x120d32c0, 0xc000c17980}, {0x120ac350, 0xc000d81d60})                                              
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.119.0/exporterhelper/internal/timeout_sender.go:53 +0xee                        
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*obsReportSender[...]).Send(0x1200f660, {0x120d32f8, 0xc000594dc0}, {0x120ac350, 0xc000d81d60})                                         
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.119.0/exporterhelper/internal/obs_report_sender.go:28 +0x21e                         
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*BaseExporter).Send(0xc000685a40, {0x120d32f8, 0xc000594dc0}, {0x120ac350, 0xc000d81d60})
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.119.0/exporterhelper/internal/base_exporter.go:129 +0xa4                             
go.opentelemetry.io/collector/exporter/exporterhelper.NewLogsRequest.func1({0x120d32f8, 0xc000594dc0}, {0xc0011a3f08?, 0xc001b03a4c?})                    
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/exporter@v0.119.0/exporterhelper/logs.go:142 +0x3c5                                        
go.opentelemetry.io/collector/consumer.ConsumeLogsFunc.ConsumeLogs(...)                                                                
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/consumer@v1.25.0/logs.go:26                                                                
go.opentelemetry.io/collector/consumer.ConsumeLogsFunc.ConsumeLogs(...)                                                               
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/consumer@v1.25.0/logs.go:26                                             
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver.(*hostEntitiesReceiver).sendEntityEvent(0xc000ad79b0, {0x120d32f8, 0xc000594dc0})                                
        /home/andrew/projects/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/receiver.go:75 +0x484                                                        
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver.(*hostEntitiesReceiver).Start(0xc000ad79b0, {0x120d16d0, 0x1a3d1860}, {0x4153a5?, 0x10?})                        
        /home/andrew/projects/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/receiver.go:33 +0xba                                                                 
go.opentelemetry.io/collector/service/internal/graph.(*Graph).StartAll(0xc000dd6360, {0x120d16d0, 0x1a3d1860}, 0xc00092ac60)                                           
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/service@v0.119.0/internal/graph/graph.go:419 +0x3fd                                             
go.opentelemetry.io/collector/service.(*Service).Start(0xc0014cbcb0, {0x120d16d0, 0x1a3d1860})                                                           
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/service@v0.119.0/service.go:267 +0x51a                                                    
go.opentelemetry.io/collector/otelcol.(*Collector).setupConfigurationComponents(0xc0007810e0, {0x120d16d0, 0x1a3d1860})                
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/otelcol@v0.119.0/collector.go:231 +0xd25                           
go.opentelemetry.io/collector/otelcol.(*Collector).Run(0xc0007810e0, {0x120d16d0, 0x1a3d1860})                                       
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/otelcol@v0.119.0/collector.go:285 +0x6b                                 
go.opentelemetry.io/collector/otelcol.NewCommand.func1(0xc0004cc308, {0x10996e15?, 0x7?, 0x1098af58?})                           
        /home/andrew/go/pkg/mod/go.opentelemetry.io/collector/otelcol@v0.119.0/command.go:36 +0x13c                          
github.com/spf13/cobra.(*Command).execute(0xc0004cc308, {0xc0000b7ea0, 0x2, 0x2})                                          
        /home/andrew/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0x10f4                                          
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004cc308)                                                                    
        /home/andrew/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x656                                       
github.com/spf13/cobra.(*Command).Execute(0xc0004cc308)                                                                    
        /home/andrew/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041 +0x27                                        
main.runInteractive({0x10dc7ce8, {{0x109cd5ab, 0xe}, {0x10bbc707, 0x3b}, {0x109b2886, 0xb}}, 0x0, {{{0x0, 0x0, ...}, ...}}, ...})                                                                       
        /home/andrew/projects/opentelemetry-collector-contrib/cmd/otelcontribcol/main.go:61 +0x5d                                                                               
main.run(...)                                                                                                                                                           
        /home/andrew/projects/opentelemetry-collector-contrib/cmd/otelcontribcol/main_others.go:10                                                             
main.main()                                                                                                                                               
        /home/andrew/projects/opentelemetry-collector-contrib/cmd/otelcontribcol/main.go:54 +0x778                                                       
exit status 2
```

### Additional context

_No response_

--- a/.chloggen/37701-fix-es-exporter-scope-attr-panic.yaml
+++ b/.chloggen/37701-fix-es-exporter-scope-attr-panic.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: elasticsearchexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Fix panic when encoding non-string scope attributes."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [37701]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/exporter/elasticsearchexporter/model.go
+++ b/exporter/elasticsearchexporter/model.go
@@ -449,11 +449,12 @@ func durationAsMicroseconds(start, end time.Time) int64 {
 
 func scopeToAttributes(scope pcommon.InstrumentationScope) pcommon.Map {
 	attrs := pcommon.NewMap()
+
+	scope.Attributes().CopyTo(attrs)
+
 	attrs.PutStr("name", scope.Name())
 	attrs.PutStr("version", scope.Version())
-	for k, v := range scope.Attributes().AsRaw() {
-		attrs.PutStr(k, v.(string))
-	}
+
 	return attrs
 }
 

--- a/exporter/elasticsearchexporter/model_test.go
+++ b/exporter/elasticsearchexporter/model_test.go
@@ -233,14 +233,19 @@ func mockResourceLogs() plog.ResourceLogs {
 func TestEncodeAttributes(t *testing.T) {
 	t.Parallel()
 
-	logRecord := plog.NewLogRecord()
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().Attributes().PutStr("keyStr", "val str")
+	scopeLogs.Scope().Attributes().PutInt("keyInt", 42)
+	logRecord := scopeLogs.LogRecords().AppendEmpty()
 	err := logRecord.Attributes().FromRaw(map[string]any{
 		"s": "baz",
 		"o": map[string]any{
 			"sub_i": 19,
 		},
 	})
-	require.NoError(t, err)
+	require.NoError(t, err, "failed to set attributes on logRecord")
 
 	tests := map[string]struct {
 		mappingMode MappingMode
@@ -253,6 +258,8 @@ func TestEncodeAttributes(t *testing.T) {
 			  "@timestamp": "1970-01-01T00:00:00.000000000Z",
 			  "Scope.name": "",
 			  "Scope.version": "",
+			  "Scope.keyInt": 42,
+			  "Scope.keyStr": "val str",
 			  "SeverityNumber": 0,
 			  "TraceFlags": 0,
 			  "o.sub_i": 19,
@@ -266,6 +273,8 @@ func TestEncodeAttributes(t *testing.T) {
 			  "@timestamp": "1970-01-01T00:00:00.000000000Z",
 			  "Scope.name": "",
 			  "Scope.version": "",
+			  "Scope.keyInt": 42,
+			  "Scope.keyStr": "val str",
 			  "SeverityNumber": 0,
 			  "TraceFlags": 0,
 			  "Attributes.o.sub_i": 19,
@@ -280,6 +289,8 @@ func TestEncodeAttributes(t *testing.T) {
 			  "agent": {
 			    "name": "otlp"
 			  },
+			  "keyInt": 42,
+			  "keyStr": "val str",
 			  "o": {
 			    "sub_i": 19
 			  },
@@ -295,8 +306,8 @@ func TestEncodeAttributes(t *testing.T) {
 
 			var buf bytes.Buffer
 			err = encoder.encodeLog(encodingContext{
-				resource: pcommon.NewResource(),
-				scope:    pcommon.NewInstrumentationScope(),
+				resource: resourceLogs.Resource(),
+				scope:    scopeLogs.Scope(),
 			}, logRecord, elasticsearch.Index{}, &buf)
 			require.NoError(t, err)
 			require.JSONEq(t, test.want, buf.String())

